### DISCUSSION TITLE
ほしいものリスト

### DISCUSSION THREAD

**Discussion by CLRafaelR** [CLRafaelR/lingdocdown#3] [Ideas]
@kishiyamat さんから頂きました，欲しいものリストです

> ほしいものです↓
> 
> - 「言語学向けの文書の生成を支援する」ということなので、手っ取り早く「これをこうしたら論文ができる」みたいなテンプレート
> - おすすめのディレクトリ構成

https://github.com/CLRafaelR/lingdocdown/discussions/1#discussioncomment-329735

**Comment by kishiyamat** [comment]
なんか「こんなレポジトリにしたい」みたいなサンプルとかありますかね。例えば、心理言語学者向けのpandoc/rmdのテンプレートみたいなレポジトリがあればそれを真似して言語学者向けにtweakするみたいな

**Reply by CLRafaelR** [reply]
# ディレクトリ構成

個人的には，下記のようにした方が，ユーザである人間にとって把握しやすいディレクトリ構成かなと思います。

wroking_directory_with_Rproj/
　├ parent.Rmd
　├ childs/
　│　├ child1.Rmd
　│　├ child2.Rmd
　│　├ ...
　│　└ childN.Rmd
　├ data/  
　│　├ csv/
　│　├ mp4/
　│　├ ...
　│　└ ext/
　└ figures/
　　　├ pdf/
　　　│　├ figure1.pdf
　　　│　├ figure2.pdf
　　　│　├ ...
　　　│　└ figureN.pdf
　　　└ png/
　　　　　├ figure1.png
　　　　　├ figure2.png
　　　　　├ ...
　　　　　└ figureN.png
  
このようなディレクトリ構成で問題なくRmdがknitできるようにするには，下記の設定が必要です。

# 親ファイル内のsetupチャンクで，パスの起点をRprojフォルダにするよう設定

[サブディレクトリで管理している Rmarkdown のデフォルトパスを設定する - shoei05](https://plaza.umin.ac.jp/shoei05/wp/index.php/2020/11/02/239/)

    ```{r setup, include=FALSE}
    knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
    ```

**Reply by kishiyamat** [reply]
あー僕もそんな感じでですね。GitHub で Template Repository  っていうのがあるそうなので (https://dev.classmethod.jp/articles/github-template-repository/) それを新しく作って、このリポジトリはそういうのを使って実際に論文執筆をすすめる方法の説明みたいなポジションもありかもしれません。

**Reply by kishiyamat** [reply]
もちろん、このリポジトリをテンプレート化して編集すればすぐに論文になるようなものにするのもありかと

**Reply by CLRafaelR** [reply]
なるほどなるほど，レポジトリそのものをテンプレート化できるのですね…！

恐らく現段階では，Rmarkdownの基礎的に使い方に注力した方がよいと思っています。今回のチュートリアルに参加なさった方も，Rmarkdownの基礎的に使い方を知りたい方が多かったからです。ですので，リポジトリのテンプレート化は，一旦論文本体のテンプレートが仕上がってからの方がいいかなと思います！

**Reply by CLRafaelR** [reply]
レポジトリをテンプレート化したときに，権利関係をどう定義するかも勉強しないといけませんよね（CC-BYをテンプレート化したレポジトリにも適用したいが，リポジトリをテンプレート化するとREADMEやLICENSEがデフォルトで設定できなくなるのをどうしたら解決できるか等）。ですので，それなりに時間もかかるかなと思います。

**Comment by CLRafaelR** [comment]
[ディレクトリ構成図を書くときに便利な記号 - Qiita](https://qiita.com/paty-fakename/items/c82ed27b4070feeceff6)

**Reply by kishiyamat** [reply]
WSL で tree コマンド使うのも早い（知ってたらすみません）

**Reply by CLRafaelR** [reply]
いえいえ，知りませんでした！なるほど，既存のフォルダ構成を出力するにはいいですね！

WindowsにもTreeがあるみたいです！
http://office-qa.com/win/win186.htm

**Reply by kishiyamat** [reply]
おぉ、やっぱり便利ですね。
